File: server/src/cv/cv_model.go
package cv

type CurriculumVitae struct {
	Basics       Basics         `json:"basics"`
	Work         []Work         `json:"work"`
	Volunteer    []Volunteer    `json:"volunteer"`
	Education    []Education    `json:"education"`
	Awards       []Awards       `json:"awards"`
	Publications []Publications `json:"publications"`
	Skills       []Skills       `json:"skills"`
	Languages    []Languages    `json:"languages"`
	Interests    []string       `json:"interests"`
	References   []References   `json:"references"`
}

type Location struct {
	City        string `json:"city"`
	Region      string `json:"region"`
	CountryCode string `json:"countryCode"`
}

type Profiles struct {
	Network  string `json:"network"`
	Username string `json:"username"`
	URL      string `json:"url"`
}

type Basics struct {
	Name     string     `json:"name"`
	Label    string     `json:"label"`
	Picture  string     `json:"picture"`
	Email    string     `json:"email"`
	Phone    string     `json:"phone"`
	Website  string     `json:"website"`
	Summary  string     `json:"summary"`
	Location Location   `json:"location"`
	Profiles []Profiles `json:"profiles"`
}

type Work struct {
	Company    string   `json:"company"`
	Position   string   `json:"position"`
	Website    string   `json:"website"`
	StartDate  string   `json:"startDate"`
	EndDate    string   `json:"endDate"`
	Summary    string   `json:"summary"`
	Highlights []string `json:"highlights"`
}

type Volunteer struct {
	Organization string   `json:"organization"`
	Position     string   `json:"position"`
	Website      string   `json:"website"`
	StartDate    string   `json:"startDate"`
	EndDate      string   `json:"endDate"`
	Summary      string   `json:"summary"`
	Highlights   []string `json:"highlights"`
}

type Education struct {
	Institution string   `json:"institution"`
	Area        string   `json:"area"`
	StudyType   string   `json:"studyType"`
	StartDate   string   `json:"startDate"`
	EndDate     string   `json:"endDate"`
	Gpa         string   `json:"gpa"`
	Courses     []string `json:"courses"`
}

type Awards struct {
	Title   string `json:"title"`
	Date    string `json:"date"`
	Awarder string `json:"awarder"`
	Summary string `json:"summary"`
}

type Publications struct {
	Name        string `json:"name"`
	Publisher   string `json:"publisher"`
	ReleaseDate string `json:"releaseDate"`
	Website     string `json:"website"`
	Summary     string `json:"summary"`
}

type Skills struct {
	Name     string   `json:"name"`
	Level    string   `json:"level"`
	Keywords []string `json:"keywords"`
}

type Languages struct {
	Language string `json:"language"`
	Fluency  string `json:"fluency"`
}

type References struct {
	Name      string `json:"name"`
	Reference string `json:"reference"`
}

File: portfolio_next/src/model/CVModel.ts
import { Document } from '@contentful/rich-text-types';

import { Entries, Entry, Media } from './Includes';

export type CVProps = {
  cv: CV;
};

export type CV = {
  about: Entry<About>;
  work: Entries<Work>;
  education: Entries<Education>;
  skills: Entry<Skills>;
  projects: Entries<Project>;
  readingList: Entries<Book>;
};

export type About = {
  name: string;
  label: string;
  email: string;
  phone: string;
  website: string;
  about: Document;
  interests: Document;
  images: Media[];
  profiles: Profile[];
};

export type Work = {
  position: string;
  company: string;
  website: string;
  startDate: Date;
  endDate: Date;
  highlights: string[];
  summary: Document;
  logo: Media;
  images: Media[];
  hideFromCv: boolean;
};

export type Education = {
  institution: string;
  link: string;
  area: string;
  studyType: string;
  startDate: Date;
  endDate: Date;
  grade: string;
  summary: Document;
  images: Media[];
};

export type Skills = {
  summary: Document;
  currentSummary: Document;
  current: Skill[];
  favoriteSummary: Document;
  favorite: Skill[];
  usedSummary: Document;
  used: Skill[];
};

export type Skill = {
  name: string;
  level: number;
  icon: DevIcon;
};

export type Project = {
  rank: number;
  name: string;
  link: string;
  image: Media;
  summary: Document;
  tags: string[];
  icon: DevIcon;
};

export enum BookProgress {
  PRIORITY = 'PRIORITY',
  NOT_STARTED = 'NOT_STARTED',
  READING = 'READING',
  COMPLETED = 'COMPLETED',
}

export type Book = {
  title: string;
  cover: Media;
  author: string;
  progress: BookProgress;
  amazonLink?: string;
};

export type Profile = {
  url: string;
  brand: DevIcon;
};

export type DevIcon = {
  name: string;
  icon: string;
  iconImage?: Media;
};
